gravitee-am-test/specs/management/password-history.jest.spec.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from "cross-fetch";
+import * as faker from 'faker';
+import {afterAll, beforeAll, expect, jest} from "@jest/globals";
+import {createDomain, deleteDomain, patchDomain, startDomain} from "@management-commands/domain-management-commands";
+import {buildCreateAndTestUser, resetUserPassword} from "@management-commands/user-management-commands";
+
+import {requestAdminAccessToken} from "@management-commands/token-management-commands";
+import {ResponseError} from "../../api/management/runtime";
+
+global.fetch = fetch;
+jest.setTimeout(200000)
+
+let accessToken;
+let domain;
+let user;
+
+beforeAll(async () => {
+    const adminTokenResponse = await requestAdminAccessToken();
+    accessToken = adminTokenResponse.body.access_token;
+    domain = await createDomain(accessToken, "domain-ph-users", faker.company.catchPhraseDescriptor()).then(async createdDomain => {
+        return await startDomain(createdDomain.id, accessToken);
+    });
+});
+
+describe("Testing password history...", () => {
+    describe("when password history is enabled", () => {
+
+        beforeAll(async () => {
+            domain = await patchDomain(domain.id, accessToken, {
+                "passwordSettings": {
+                    "passwordHistoryEnabled": true,
+                    "oldPasswords": 3
```

Comment:
In order to fully validate the feature, maybe we should also test that after 3 reset password the first one can be reused? wdyt?

Reply:
Agree. but there isn't really time at this stage to add anything else to the test (given that I'm on support and the only person fully in the team today).

Reply:
Done.